bot/src/discord/schedule_helper.js

```diff
@@ -0,0 +1,306 @@
+const dbConnection = require('kotoba-node-common').database.connection;
+const WordScheduleModel = require('kotoba-node-common').models.createWordSchedulesModel(dbConnection);
+const showRandomWord = require('./show_random_word.js');
+
+const frequencyCheck = 60 * 1000;
+
+const statusConstants = {
+  running: 'running',
+  paused: 'paused',
+};
+
+/**
+ * Starts the interval
+ * @param {Monochrome} monochrome
+ * @param {boolean} firstCall
+ */
+async function setTimer(monochrome, firstCall) {
+  let now = (new Date()).getTime();
+  // The first time this function is called we don't want to run the command,
+  // just set the timer so it runs in sync with the frequencyCheck value
+  if (!firstCall) {
+    const pendingSchedules = await WordScheduleModel.find({ status: statusConstants.running });
+    pendingSchedules.forEach(async (schedule) => {
+      // We check if the schedule has reached the start date and if it's in sync
+      // with the frequency, with an error margin of 0.1s
+      if (now >= schedule.start && (now - schedule.start) % schedule.frequency < 100) {
```

Comment:
This math feels suspicious to me. I would suggest saving the `lastSent` into the database and doing:

```js
const due = now >= schedule.start
  && (!schedule.lastSent || (now - schedule.lastSent) > schedule.frequency);